src/daemon/daemon.cpp

```diff
@@ -136,7 +136,14 @@ bool t_daemon::run(bool interactive)
   {
     throw std::runtime_error{"Can't run stopped daemon"};
   }
-  tools::signal_handler::install(std::bind(&daemonize::t_daemon::stop_p2p, this));
+
+  bool stop = false;
```

Comment:
Shouldn't this be an atomic bool?

Reply:
We don't really care with the usage which is made of it. I don't really imagine bools being non atomic anyway on a sane platform.

Reply:
It's not the atomicity that matters, it's the visibility -- there's no guarantee that the second thread will see the changed value.

Reply:
(Yes, technically you just need a memory fence, not an atomic, but the atomic gives you that and is easier :))

Reply:
Do you mean right after it (we don't really care), or ever (in which case, the world would fall over in many other places) ?

Reply:
I *think* it's theoretically "ever," and practically "as long as the old value remains in L1 cache." I'm not 100% sure, maybe @hyc knows for sure?

Reply:
Is there any downside to making it atomic btw?

Reply:
Just a bit more verbose.

Reply:
Yeah, that's kind of why I suggest just using atomic instead of an explicit memory fence.

Without either, though, the writing thread has no indication that it needs to flush the dirty cache line containing stop=true.

Reply:
(Eventually, it will happen anyway once another thread is scheduled on the same CPU and evicts that line to use it for something else. I don't think it's correct to rely on that, though; one can easily imagine edge cases where it doesn't happen for a while.)

Reply:
(I could still be wrong about this, it's been a little while. But I'm like 95% sure)